pkg/http/handler/timeout.go

```diff
@@ -49,15 +52,19 @@ type timeToFirstByteTimeoutHandler struct {
 // https://golang.org/pkg/net/http/#Handler.
 //
 // The implementation is largely inspired by http.TimeoutHandler.
-func NewTimeToFirstByteTimeoutHandler(h http.Handler, msg string, timeout time.Duration) http.Handler {
-	return &timeToFirstByteTimeoutHandler{
-		handler: h,
-		body:    msg,
-		timeout: timeout,
+func NewTimeoutHandler(h http.Handler, msg string, firstByteTimeout time.Duration, idleTimeout time.Duration) http.Handler {
+	if idleTimeout < 0 {
+		idleTimeout = math.MaxInt64 * time.Nanosecond
+	}
```

Comment:
Oh I see, this is how you're "disabling" the timeout. See comment below, I'd hope we can actually work around even allocating a timer in this case.

Reply:
I would vote to use zero value for disabling the timeout, as this is also used in [Golang http implementation](https://pkg.go.dev/net/http#pkg-types). Zero is used for other features  in Serving: 
```
Setting a value of zero
    # disables this throttling and lets through as many requests as
    # the pod receives.
    container-concurrency: "0"
```
This way we keep semantics the same.

Reply:
Added a new struct optionalTimer (try it make it a bit more elegant) to handle this "timer disabled" situation